base_import_odoo/models/import_odoo_database.py

```diff
@@ -0,0 +1,472 @@
+        return ImportContext(*(self[:-1] + (field_context(*args),)))
+
+
+field_context = namedtuple(
+    'field_context', ['record_model', 'field_name', 'record_id'],
+)
+
+
+mapping_key = namedtuple('mapping_key', ['model_name', 'remote_id'])
+
+
+dummy_instance = namedtuple(
+    'dummy_instance', ['model_name', 'field_name', 'remote_id', 'dummy_id'],
+)
+
+
+class ImportOdooDatabase(models.Model):
+    _name = 'import.odoo.database'
+    _description = 'An Odoo database to import'
+
+    url = fields.Char(required=True)
+    database = fields.Char(required=True)
+    user = fields.Char(default='admin', required=True)
+    password = fields.Char(default='admin')
+    import_line_ids = fields.One2many(
+        'import.odoo.database.model', 'database_id', string='Import models',
+    )
+    import_field_mappings = fields.One2many(
+        'import.odoo.database.field', 'database_id', string='Field mappings',
+    )
+    cronjob_id = fields.Many2one(
+        'ir.cron', string='Import job', readonly=True, copy=False,
+    )
+    cronjob_running = fields.Boolean(compute='_compute_cronjob_running')
+    status_data = fields.Serialized('Status', readonly=True, copy=False)
+    status_html = fields.Html(
+        compute='_compute_status_html', readonly=True, sanitize=False,
+    )
+
+    @api.multi
+    def action_import(self):
+        """Create a cronjob to run the actual import"""
+        self.ensure_one()
+        if self.cronjob_id:
+            return self.cronjob_id.write({
+                'numbercall': 1,
+                'doall': True,
+                'active': True,
+            })
+        return self.write({
+            'cronjob_id': self._create_cronjob().id,
+        })
+
+    @api.multi
+    def _run_import(self, commit=True, commit_threshold=100):
+        """Run the import as cronjob, commit often"""
+        self.ensure_one()
+        if not self.password:
+            return
+        # model name: [ids]
+        remote_ids = {}
+        # model name: count
+        remote_counts = {}
+        # model name: count
+        done = {}
+        # mapping_key: local_id
+        idmap = {}
+        # mapping_key: local_id
+        # this are records created or linked when we need to fill a required
+        # field, but the local record is not yet created
+        dummies = {}
+        # model name: [local_id]
+        # this happens when we create a dummy we can throw away again
+        to_delete = {}
+        # dummy_instance
+        dummy_instances = []
+        remote = self._get_connection()
+        self.write({'password': False})
+        if commit and not tools.config['test_enable']:
+            # pylint: disable=invalid-commit
+            self.env.cr.commit()
+        for model_line in self.import_line_ids:
+            model = model_line.model_id
+            remote_ids[model.model] = remote.search(
+                model.model,
+                tools.safe_eval(model_line.domain) if model_line.domain else []
```

Comment:
A common need is to fetch only the records change since the last successful sync. It would be nice to add support for that.

Reply:
sure. but I'd also like a mode where we can update all again (should look at the modification dates then which one wins), then we can use this for two way sync